### PR TITLE
Refactor coinswap

### PR DESCRIPTION
**Pull request by dreamer-zq**

### CODE REVIEW

File: modules/coinswap/keeper/keeper.go [hunk 1]
```diff
@@ -105,30 +105,31 @@ func (k Keeper) AddLiquidity(ctx sdk.Context, msg *types.MsgAddLiquidity) (sdk.C
...
 	var depositToken sdk.Coin
 	var standardCoin = sdk.NewCoin(standardDenom, msg.ExactStandardAmt)
 
+	poolId := types.GetPoolId(msg.MaxToken.Denom)
+	pool, has := k.GetPool(ctx, poolId)
+
 	// calculate amount of UNI to be minted for sender
 	// and coin amount to be deposited
-	if liquidity.IsZero() {
+	if !has {
```
Comment (zhangyelong) [inline modules/coinswap/keeper/keeper.go:117]: How about rename `has` to `exists`?

File: modules/coinswap/keeper/pool.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	nextSequence := k.getNextPoolSequence(ctx)
+	lptDenom := types.GetLptDenom(nextSequence)
+	pool := &types.Pool{
+		Id:                types.GetPoolId(counterpartyDenom),
+		StandardDenom:     k.GetStandardDenom(ctx),
+		CounterpartyDenom: counterpartyDenom,
+		EscrowAddress:     types.GetReservePoolAddr(lptDenom).String(),
+		LptDenom:          lptDenom,
+	}
+	k.setNextPoolSequence(ctx, nextSequence+1)
```
Comment (zhangyelong) [inline modules/coinswap/keeper/pool.go:23]: I would prefer to use `increase` instead of `set`

File: modules/coinswap/migrations/v1110/migrate.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+
+	coinswapkeeper "github.com/irisnet/irismod/modules/coinswap/keeper"
+	coinswaptypes "github.com/irisnet/irismod/modules/coinswap/types"
+)
+
+func Migrate(ctx sdk.Context, k coinswapkeeper.Keeper, bk bankkeeper.Keeper, ak authkeeper.AccountKeeper) error {
+	// 1. Query all current liquidity tokens
+	var ltpDenoms []string
```
Comment (zhangyelong) [inline modules/coinswap/migrations/v1110/migrate.go:17]: lptDenom ?

File: modules/coinswap/keeper/pool.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+// GetLptDenomFromDenoms returns the liquidity pool token denom for the provided denominations.
+func (k Keeper) GetLptDenomFromDenoms(ctx sdk.Context, denom1, denom2 string) (string, error) {
+	if denom1 == denom2 {
+		return "", types.ErrEqualDenom
+	}
+
+	standardDenom := k.GetStandardDenom(ctx)
+	if denom1 != standardDenom && denom2 != standardDenom {
+		return "", sdkerrors.Wrap(types.ErrNotContainStandardDenom, fmt.Sprintf("standard denom: %s,denom1: %s,denom2: %s", standardDenom, denom1, denom2))
```
Comment (chengwenxi) [inline modules/coinswap/keeper/pool.go:84]:
```suggestion
		return "", sdkerrors.Wrap(types.ErrNotContainStandardDenom, fmt.Sprintf("standard denom: %s, denom1: %s, denom2: %s", standardDenom, denom1, denom2))
```

File: modules/coinswap/keeper/swap_test.go [hunk 5]
```diff
@@ -49,7 +49,7 @@ func (suite *TestSuite) TestGetInputPrice() {
 	for _, tcase := range datas {
 		data := tcase.data
 		actual := keeper.GetInputPrice(data.delta, data.x, data.y, data.fee)
-		_, _ = fmt.Printf("expect:%s,actual:%s", tcase.expect.String(), actual.String())
+		_, _ = fmt.Println("expect:", tcase.expect.String(), "actual:", actual.String())
```
Comment (chengwenxi) [inline modules/coinswap/keeper/swap_test.go:52]: Remove this self-debug code.

File: modules/coinswap/keeper/swap_test.go [hunk 6]
```diff
@@ -71,7 +71,7 @@ func (suite *TestSuite) TestGetOutputPrice() {
 	for _, tcase := range datas {
 		data := tcase.data
 		actual := keeper.GetOutputPrice(data.delta, data.x, data.y, data.fee)
-		_, _ = fmt.Printf("expect:%s,actual:%s", tcase.expect.String(), actual.String())
+		_, _ = fmt.Printf("expect: %s,actual: %s", tcase.expect.String(), actual.String())
```
Comment (chengwenxi) [inline modules/coinswap/keeper/swap_test.go:74]: ditto.

File: modules/coinswap/migrations/v1110/migrate.go [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+			return err
+		}
+	}
+
+	// 4. Traverse all accounts and modify the old liquidity token to the new liquidity token
+	ak.IterateAccounts(ctx, func(account authtypes.AccountI) (stop bool) {
+		balances := bk.GetAllBalances(ctx, account.GetAddress())
+		for _, ltpDenom := range ltpDenoms {
+			amount := balances.AmountOf(ltpDenom)
+			if sdk.ZeroInt().Equal(amount) {
```
Comment (chengwenxi) [inline modules/coinswap/migrations/v1110/migrate.go:42]:
```suggestion
			if amount.IsZero() {
```

**Review by chengwenxi — CHANGES_REQUESTED** [review]

**Review by zhangyelong — APPROVED** [review]

**Review by chengwenxi — APPROVED** [review]

**Pull request merged**